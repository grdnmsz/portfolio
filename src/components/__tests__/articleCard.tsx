import React from 'react'
import renderer from 'react-test-renderer'

import { ArticleCard } from '../'

describe('ArticleCard', () => {
  const props = {
    frontmatter: {
      title: 'a-title',
      date: '10-10-2020',
      description: 'some description to fill the gaps',
      category: 'web / foo'
    },
    fields: {
      slug: '#a-title'
    }
  }

  it('should render correctly', () => {
    const tree = renderer.create(<ArticleCard {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
