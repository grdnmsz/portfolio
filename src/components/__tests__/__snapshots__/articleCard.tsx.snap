// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard should render correctly 1`] = `
<div
  className="max-w-lg my-20 "
>
  <div
    className="bg-orange-700 max-w-max mb-2 px-4 py-1 rounded-2xl text-xs"
  >
    web / foo
  </div>
  <h3
    className="sm:text-2xl text-xl"
  >
    a-title
  </h3>
  <div
    className="text-gray-500 italic"
  >
    some description to fill the gaps
  </div>
  <div
    className="flex items-end relative"
  >
    <a
      className="hover:text-teal-100"
      href="#a-title"
    >
      <div
        className="border hover:text-teal-100 hover:border-teal-100 max-w-max mt-2 p-2 rounded"
      >
        Read article
      </div>
    </a>
    <span
      className="absolute right-0 text-xs sm:text-base"
    >
      🗓️ 
      10-10-2020
    </span>
  </div>
</div>
`;
